sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -48,7 +48,8 @@ public virtual async ValueTask<AccessToken> AuthenticateAsync(bool async, TokenR
 
         private static ManagedIdentitySource SelectManagedIdentitySource(ManagedIdentityClientOptions options)
         {
-            return AppServiceV2017ManagedIdentitySource.TryCreate(options) ??
+            return AppServiceV2019ManagedIdentitySource.TryCreate(options) ??
+                AppServiceV2017ManagedIdentitySource.TryCreate(options) ??
```

Comment:
There is some nuance here. I think with adding support for the 2019 protocol we need to consider both `AppServiceV2019ManagedIdentitySource` and `AppServiceV2017ManagedIdentitySource` after `ServiceFabricManagedIdentitySource` otherwise we will break Service Fabric  managed identity support. This is because `ServiceFabricManagedIdentitySource` should be used if the environment variables `IDENTITY_ENDPOINT`, `IDENTITY_HEADER` _AND_ `IDENTITY_SERVER_THUMBPRINT` are set. If we consider `AppServiceV2019ManagedIdentitySource` before it, we will incorrectly identify the Service Fabric host as the App Service host. 

Also `AppServiceV2017ManagedIdentitySource` must be considered after `AppServiceV2019ManagedIdentitySource` (as you currently are doing) because App Service hosts which support the 2019 protocol will have both pairs of `IDENTITY_ENDPOINT`, `IDENTITY_HEADER` _AND_ `MSI_ENDPOINT`, `MSI_SECRET` set, so in this case we want to prefer the newer protocol.

We should add comment to the rational behind there ordering here, and make sure we have tests covering these cases.

Reply:
In this case, we are handling it properly in the 2019 provider, but I think the ordering fix is better because it doesn't require the V2019 source from having to check the `IDENTITY_SERVER_THUMBPRINT` environment variable.